Changelog: DM-3411 Loriot-Agent output user missing Loriot Admin Role (#1973)
SoftwareAG/c8y-docs: content/change-logs/device-management/loriot-1-0-0-Assign-role-to-uplink-output-user.md

---
date:

> title: Assign Loriot Admin Role to the Loriot agent output user for existing and new devices.   

Reviewer: ```suggestion
title: Assign Loriot admin permissions to the Loriot agent output user for existing and new devices   
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-ycWx1InI9
    label: loriot-agent
ticket: DM-3411
version: 1.0.0
---

> Newly established Loriot connections failed initially because the user account created for sending uplink requests from devices lacked the necessary permissions. This issue has been resolved by granting the required permissions to both existing and new connections.

Reviewer: the required permissions > shouldn´t this be more precise? The title mentions admin permissions

Author: I can change this to required admin permissions

Author: Hope the changes are fine now

> Newly established LORIOT connections failed initially because the user account created for sending uplink requests from devices lacked the necessary permissions. This issue has been resolved by granting the required permissions to both existing and new connections.

Author: @l2p- @aaronraab I think the sentence should now be fine. 

We could also use the below (this was a rephrase of the sentence @l2p- had suggested, by Gemini), if this isnt as per expectation:

The initial LORIOT setup malfunctioned due to insufficient permissions for the user created to transmit device data (uplink requests) to the platform. This issue has been resolved by granting the necessary permissions to both existing and new connections.

> Newly established LORIOT connections failed initially because the user account created for sending uplink requests from devices lacked the necessary permissions. This issue has been resolved by granting the required permissions to both existing and new connections.

Reviewer: ```suggestion
Newly established Loriot connections failed initially because the user account created for sending uplink requests from devices lacked the necessary permissions. This issue has been resolved by granting the required permissions to both existing and new connections.
```

Reviewer: In the title you are more precise on which permissions have been added. I would better add this information in the detailed description.

---
date:
title: Assign Loriot Admin Role to the Loriot agent output user for existing and new devices.   
product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-ycWx1InI9
    label: loriot-agent
ticket: DM-3411
version: 1.0.0
---

> The Loriot agent output user was missing the Loriot Admin Role, due to which devices were unable to send uplink messages to cumulocity. 
> This issue has now been addressed. Existing devices as well as new devices will now be able to send uplink messages, as the output user will be assigned the role. 

Reviewer 2: Users don't know what a "Loriot agent output user" is and why it would need that role. The part about devices not being able to send uplink is also misleading because they can but Loriot cannot forward them to us. We need to describe things in a way that uses terminology and functionality that is known to users.
```suggestion
Newly set up LORIOT connections did not work properly because during the setup process because the user created for LORIOT to send uplink requests from devices to the platform did not have sufficient permissions to do so. This has been addressed by adding the required permissions to existing connections and new ones.
```

Reviewer 3: This is really hard to read as you are using because 2 times -> 
Newly set up Loriot connections did not work properly because  the user that is being created during the setup process and used for Loriot to send uplink requests from devices to the platform did not have sufficient permissions to do so. This has been addressed by adding the required permissions to existing connections as well as new ones.

Author: Change has been made now
